src/UI/Utils/Action.elm

```diff
@@ -0,0 +1,60 @@
+module UI.Utils.Action exposing
+    ( Config, Action(..)
+    , map, mapIcon
+    )
+
+{-| Describes a clickable element for components with customizable clickable action.
+
+
+# Building
+
+@docs Config, Action
+
+-}
+
+import UI.Icon exposing (Icon)
+import UI.Link exposing (Link)
+
+
+{-| `Config` assembles the required configuration for having an action element.
+
+    { label = "Create new element"
+    , icon = Icon.add
+    , action =
+        DialogMsg.OpenElementCreation
+            |> Msg.ForDialog
+            |> Action.DispatchMsg
+    }
+
+-}
+type alias Config msg =
+    { action : Action msg
+    , icon : String -> Icon
```

Comment:
will it always have an icon?

Reply:
This one is a great question. Previously I've asked @jouderianjr  the same, and his answer was: it doesn't make much sense to not having it.
While my opinion is: having a clickable text, or any other non-icon component, will just add enough abstraction that would make more sense to use elm-ui directly.

Reply:
My statement is a bit out of context here :o hahaha. But when I said that, I was talking more specifically to the StackChild, not exactly in general.

But I'm not sure if it makes sense to support any other thing here 🤔  we don't have right now an example of that because in the end this Module is used by other modules and not in elm-ui directly.